nixos/modules/services/hardware/nbfc-linux.nix

```diff
@@ -0,0 +1,61 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.nbfc-linux;
+
+  nbfcConfigList = splitString "\n" (
+    removeSuffix "\n" (
+      builtins.readFile (
+        pkgs.runCommand "nbfc-config-list" { } ''
+          ${pkgs.nbfc-linux}/bin/nbfc config --list > $out
+        ''
+      )
+    )
+  );
+in
+{
+  options.services.nbfc-linux = {
+    enable = lib.mkEnableOption "NBFC: NoteBook FanControl service";
+
+    package = lib.mkPackageOption pkgs "nbfc-linux" { default = "nbfc-linux "; };
+
+    configName = mkOption {
+      type = types.nullOr types.str;
+      description = "The NBFC configuration name to use. Obtain the list via `nbfc config --list`";
+      default = null;
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    assertions = [
+      {
+        assertion = (cfg.configName == null) || (builtins.elem cfg.configName nbfcConfigList);
+        message = "nbfc-linux: invalid configName: ${cfg.configName}";
+      }
+    ];
+
+    environment.systemPackages = [ pkgs.nbfc-linux ];
```

Comment:
Does this need to be installed globally by default?

Reply:
i'm new to nixos - is there a better way of ensuring the `nbfc-linux` package is installed? this service depends on that package

Reply:
i think it need to be in systempackages for the `nbfc` exe to provided to the userspace? nice to be able to run `nbfc status`

Reply:
If thats a use case, then it's valid to install it, sure